### PR TITLE
canfdtest: Add message length support

### PR DESCRIPTION
**Pull request by RICCIARDI-Adrien**
Added the '-s' command line option to set the frame payload length in bytes.

### CODE REVIEW

File: canfdtest.c [hunk 1]
```diff
@@ -470,6 +476,24 @@ int main(int argc, char *argv[])
...
+		printf("Message length must > 0\n");
+		return 1;
+	}
+	if (is_can_fd) {
+		if (msg_len > CANFD_MAX_DLEN) {
+			printf("Message length must be <= %d bytes for CAN FD\n", CANFD_MAX_DLEN);
+			return 1;
+		}
+	}
+	else {
```
Comment (marckleinebudde) [inline canfdtest.c:490]: Nitpick: please use kernel coding style:
`} else {`

**Review by marckleinebudde — CHANGES_REQUESTED** [review]
looks good, just a coding style nitpick.

File: canfdtest.c (hunk 1 above)
Comment (RICCIARDI-Adrien) [reply canfdtest.c]: Oops, sorry !

**Review by marckleinebudde — APPROVED** [review]

**Comment by gibsson** [comment]
Sorry I didn't have time to dig more into the issue but one of the patch from Adrien breaks the Android build:
```
external/can-utils/lib.c:575:28: error: comparison of integers of different signs: 'int' and 'unsigned long' [-Werror,-Wsign-compare]
                                if (tmp_n < 0 || tmp_n >= len - n){
                                                 ~~~~~ ^  ~~~~~~~
external/can-utils/lib.c:581:27: error: comparison of integers of different signs: 'int' and 'unsigned long' [-Werror,-Wsign-compare]
                        if (tmp_n < 0 || tmp_n >= len - n){
                                         ~~~~~ ^  ~~~~~~~
external/can-utils/lib.c:661:28: error: comparison of integers of different signs: 'int' and 'unsigned long' [-Werror,-Wsign-compare]
                                if (tmp_n < 0 || tmp_n >= len - n){
                                                 ~~~~~ ^  ~~~~~~~
external/can-utils/lib.c:667:27: error: comparison of integers of different signs: 'int' and 'unsigned long' [-Werror,-Wsign-compare]
                        if (tmp_n < 0 || tmp_n >= len - n){
                                         ~~~~~ ^  ~~~~~~~
4 errors generated.
```

**Comment by gibsson** [comment]
Oops, my jenkins logs weren't correct, it's not this pull-request but this patch that is the issue:
https://github.com/linux-can/can-utils/commit/ef853f555379babc121b90af4e8d2427aef7f2d3
Sorry for the noise Adrien ;)

**Comment by marckleinebudde** [comment]
That issue was introduced in PR https://github.com/linux-can/can-utils/pull/321.

@weichslgartner, can you create patches to fix the issue?

**Comment by marckleinebudde** [comment]
I've opened a new issue

**Pull request merged**